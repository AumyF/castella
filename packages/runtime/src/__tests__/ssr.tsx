/**
 * @jest-environment node
 */
import React, { Fragment } from "react";
import { renderToStaticMarkup, renderToString } from "react-dom/server";
import { component, ServerRenderingCollector, styledComponent } from "..";

describe("SSR", () => {
  it("input", () => {
    const Hello = component({
      shadowHtml: `
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <div><slot></slot></div>
      `,
      slots: [],
      name: "wc-test-foo",
      classicSSR: true,
    });
    const Hello2 = styledComponent({
      ssr: true,
      elementName: "input",
      className: "test-hello2",
      css: ".test-hello2 {color: red}",
    });

    const sheet = new ServerRenderingCollector();
    const str = renderToString(
      sheet.wrap(
        <Hello>
          <Hello2 />
        </Hello>
      )
    );

    expect(str).toMatchSnapshot();
    expect(
      renderToStaticMarkup(<Fragment>{sheet.getHeadElements()}</Fragment>)
    ).toMatchSnapshot();
  });
});
