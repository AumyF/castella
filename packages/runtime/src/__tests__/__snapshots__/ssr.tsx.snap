// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR input 1`] = `"<wc-test-foo><input class=\\"test-hello2\\"/></wc-test-foo>"`;

exports[`SSR input 2`] = `"<style data-castella-styled=\\"test-hello2\\">.test-hello2 {color: red}</style><script>var __REACT_WC__={c:()=>class E extends HTMLElement{constructor(){super();this.attachShadow({mode:\\"open\\"}).appendChild(E.template.cloneNode(true))}refresh(){let c,s=this.shadowRoot;if(s){while(c=s.firstChild)s.removeChild(c);s.appendChild(E.template.cloneNode(true))}}},p:(t,d)=>{return d=document.createElement(\\"div\\"),d.insertAdjacentHTML(\\"afterbegin\\",t),t=document.createDocumentFragment(),t.appendChild(...d.childNodes),t}};customElements.define(\\"wc-test-foo\\",(E=>(E.template=__REACT_WC__.p(\\"\\\\n          <style>\\\\n            div {\\\\n              font-size: 100px;\\\\n            }\\\\n          </style>\\\\n          <div><slot></slot></div>\\\\n      \\"),E))(__REACT_WC__.c()));</script>"`;
